quest_test/test_resource_stream.py

```diff
@@ -14,60 +22,76 @@ async def workflow():
 
-            async with queue('messages', None) as messages:
-                new_message = await messages.get()
-                await phrase.set((await phrase.get()) + new_message)
+            # async with queue('messages', None) as messages:
+            #     new_message = await messages.get()
+            #     await phrase.set((await phrase.get()) + new_message)
 
             await phrase.set('all done')
 
-    history = []
-    historian = Historian('test', workflow, history)
-    wtask = historian.run()
-
-    updates = aiter(historian.stream_resources(None))
-    resources = await anext(updates)  # empty - start of workflow
-
-    resources = await anext(updates)  # phrase created
-    assert 'phrase' in resources
-    assert await resources['phrase'].value() == 'woot'
-
-    resources = await anext(updates)  # phrase.get()
-    resources = await anext(updates)  # phrase.set(big_phrase())
-    assert 'phrase' in resources
-    assert await resources['phrase'].value() == 'wootwootwoot'
-
-    resources = await anext(updates)  # messages created
-    assert 'phrase' in resources
-    assert await resources['phrase'].value() == 'wootwootwoot'
-
-    assert 'messages' in resources
-    await resources['messages'].put('quuz')
-
-    resources = await anext(updates)  # messages.get()
-    resources = await anext(updates)  # phrase.get()
-    resources = await anext(updates)  # phrase.set(+ message)
-    assert 'phrase' in resources
-    assert await resources['phrase'].value() == 'wootwootwootquuz'
-    assert 'messages' in resources
-
-    resources = await anext(updates)  # messages deleted
-    assert 'phrase' in resources
-    assert await resources['phrase'].value() == 'wootwootwootquuz'
-    assert 'messages' not in resources
-
-    resources = await anext(updates)  # phrase.set
-    assert 'phrase' in resources
-    assert await resources['phrase'].value() == 'all done'
-
-    resources = await anext(updates)  # phrase deleted
-    assert not resources  # i.e. empty
-
-    resources = await anext(updates)  # workflow complete
-
-    try:
-        resources = await anext(updates)
-        pytest.fail('Should have thrown StopAsyncIteration')
-    except StopAsyncIteration:
-        pass
-
-    await wtask
+    w_id = 'input-' + str(uuid.uuid1())
+    async with create_filesystem_manager(
+            Path('test_state'),
+            'resource_stream_test',
+            lambda workflow_id: workflow
+    ) as manager:
+        manager.start_workflow('', w_id)
+        # TODO: Discuss whether stream_resources should be async or not. (And make so in manager, historian, resources)
+        index = 0
+        async for resources in manager.stream_resources(w_id, None):
+            if index == 0:
+                manager.start_workflow('', w_id)
+            print(f'Resources: {resources}')
+            index += 1
+
+        # This isn't implemented in manager...
+        # await manager.wait_for_completion(w_id, None)
```

Comment:
We don't want this anymore. Now that the stream closes when the loop exits, we should be able to `break` from the loop and be ok.

That's a test we should try. We know that throwing an exception closes a context in the generator function. What if the client code `break`s from the loop? What if I `return` from within a loop? Does the context in the generator still get closed?